AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/DeviceOrientationHandler.java

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package com.microsoft.accessibilityinsightsforandroidservice;
+
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+public class DeviceOrientationHandler {
+  private int orientation;
+  private ArrayList<Consumer<Integer>> onOrientationChangedListeners;
+
+  public DeviceOrientationHandler(int orientation) {
+    this.orientation = orientation;
+    this.onOrientationChangedListeners = new ArrayList<>();
+  }
+
+  public void subscribe(Consumer<Integer> listener) {
+    this.onOrientationChangedListeners.add(listener);
+  }
+
+  public void setOrientation(int orientation) {
+    if (this.orientation != orientation) {
```

Comment:
nit: can flip this around by using guard clause here.

Reply:
done!